### ISSUE TITLE
`Origin` header contains path

### ISSUE THREAD

**Issue by staticfloat** [JuliaWeb/HTTP.jl#53]
The `Origin` header seems to contain the full path, which causes some CORS-checking systems to choke.
```
julia> HTTP.request(client, "GET", HTTP.URI("https://buildog.julialang.org/api/v2/forceschedulers"); verbose=true)
[HTTP - 2017-05-18T19:33:00.093]: using request options: :chunksize=>1048576, :gzip=>true, :connecttimeout=>15.0, :readtimeout=>15.0, :
tlsconfig=>MbedTLS.SSLConfig(), :maxredirects=>5, :allowredirects=>true, :forwardheaders=>false, :retries=>3
[HTTP - 2017-05-18T19:33:00.368]: checking if any existing connections to 'buildog.julialang.org' are re-usable
[HTTP - 2017-05-18T19:33:00.368]: found re-usable connection #1
[HTTP - 2017-05-18T19:33:00.406]: sending request over the wire
HTTP.Request:
"""
GET /api/v2/forceschedulers HTTP/1.1
Host: buildog.julialang.org
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8,application/json
Origin: https://buildog.julialang.org/api/v2/forceschedulers
User-Agent: HTTP.jl/0.0.0

"""
```
However, according to [Mozilla's docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin):

> The Origin request header indicates where a fetch originates from. It doesn't include any path information, but only the server name. It is sent with CORS requests, as well as with POST requests. It is similar to the Referer header, but, unlike this header, it doesn't disclose the whole path.

**Comment by quinnj** [comment]
Ah, excellent info; yeah, we should make sure to only send it w/ POST requests as well as limiting to non-path url.

**Comment by staticfloat** [comment]
Luckily, this is not a problem on the latest release of `HTTP.jl`, but on `master`, this makes it so that you cannot connect to some sites.  :(

**Comment by quinnj** [comment]
Mind giving lastest master a try to see if that fixes it? ae33677d428a3cc1d0d24199afcff8b86b712495
